control-plane/src/e3_client.rs

```diff
@@ -0,0 +1,71 @@
+use std::net::SocketAddr;
+
+use tokio::net::TcpStream;
+
+use crate::{
+    error::Result,
+    internal_dns::{self, AsyncDnsResolver},
+};
+
+pub struct E3Client {
+    socket_addr: Option<SocketAddr>,
```

Comment:
Adding some sort of TTL to the E3 IP would be good — we use [cached](https://github.com/evervault/cages/blob/main/data-plane/src/cache.rs) in the data plane